crates/astria-conductor/src/celestia/mod.rs

```diff
@@ -130,138 +140,173 @@ pub(crate) struct Reader {
+                .await
+                .wrap_err("handle to executor failed while waiting for it being initialized")
+        };
 
-        let block_verifier = BlockVerifier::new(self.sequencer_cometbft_client.clone());
+        let get_sequencer_chain_id = async {
+            get_sequencer_chain_id(self.sequencer_cometbft_client.clone())
+                .await
+                .wrap_err("failed to get sequencer chain ID")
+        };
 
-        debug!(
-            %rollup_id,
-            %initial_expected_sequencer_height,
-            %initial_celestia_height,
-            celestia_variance,
-            "setting up celestia reader",
-        );
+        try_join!(wait_for_init_executor, get_sequencer_chain_id)
+    }
+}
 
-        let latest_celestia_height = match self.latest_celestia_heights.next().await {
-            Some(Ok(height)) => height,
-            Some(Err(e)) => {
-                return Err(e).wrap_err("subscription to celestia header returned an error");
-            }
-            None => bail!("celestia header subscription was terminated unexpectedly"),
-        };
+struct RunningReader {
+    block_cache: BlockCache<ReconstructedBlock>,
 
-        debug!(
-            height = latest_celestia_height,
-            "received latest height from celestia"
-        );
+    blob_verifier: Arc<BlobVerifier>,
 
-        // XXX: This block cache always starts at height 1, the default value for `Height`.
-        let mut sequential_blocks =
-            BlockCache::<ReconstructedBlock>::with_next_height(initial_expected_sequencer_height)
-                .wrap_err("failed constructing sequential block cache")?;
-
-        let mut block_stream = ReconstructedBlocksStream {
-            track_heights: TrackHeights {
-                reference_height: initial_celestia_height.value(),
-                variance: celestia_variance,
-                last_observed: latest_celestia_height,
-                next_height: initial_celestia_height.value(),
-            },
-            // NOTE: Gives Celestia 600 seconds to respond. This seems reasonable because we need to
-            // 1. fetch all sequencer header blobs, 2. fetch the rollup blobs, 3. verify the rollup
-            // blobs.
-            // XXX: This should probably have explicit retry logic instead of this futures map.
-            in_progress: FuturesMap::new(Duration::from_secs(600), 10),
-            client: self.celestia_client.clone(),
-            verifier: block_verifier.clone(),
-            sequencer_namespace,
+    // Client to fetch heights and blocks from Celestia.
+    celestia_client: CelestiaClient,
+
+    /// The channel used to send messages to the executor task.
+    executor: executor::Handle<StateIsInit>,
+
+    /// Token to listen for Conductor being shut down.
+    shutdown: CancellationToken,
+
+    /// Tasks reconstructing Sequencer block information from Celestia blobs.
+    tasks: JoinMap<u64, eyre::Result<ReconstructedBlocks>>,
```

Comment:
```suggestion
    reconstruction_tasks: JoinMap<u64, eyre::Result<ReconstructedBlocks>>,
```

tasks is a bit vague :p